src/Raven.Server/Documents/Indexes/VectorSearch/GenerateEmbeddings.cs

```diff
@@ -0,0 +1,242 @@
+            return new VectorValue(arrayPool: null, (byte[])(object)array, (byte[])(object)array);
+        }
+        
+        if (embeddingSourceType is VectorEmbeddingType.Int8)
+        {
+            if (typeof(T) == typeof(byte))
+                return new VectorValue(arrayPool: null, (byte[])(object)array, new Memory<byte>((byte[])(object)array, 0, array.Length));
+            
+            PortableExceptions.ThrowIf<InvalidDataException>(typeof(T) != typeof(sbyte), $"Data already quantized in '{VectorEmbeddingType.Int8}' form should be of type '{typeof(sbyte)}'.");
+            var bytes = MemoryMarshal.Cast<T, byte>(array);
+            var allocator = Allocator ??= ArrayPool<byte>.Create();
+            var buffer = allocator.Rent(bytes.Length);
+            bytes.CopyTo(buffer.AsSpan());
+            
+            return new VectorValue(arrayPool: allocator, buffer, new Memory<byte>(buffer, 0, array.Length));
+        }
+        
+        switch (embeddingDestinationType)
+        {
+            case VectorEmbeddingType.Binary:
+            {
+                ReadOnlySpan<float> input;
+                if (array is float[] af)
+                    input = af;
+                else
+                    input = MemoryMarshal.Cast<T, float>(array);
+                
+                var binaryEmbedding = VectorQuantizer.ToInt1(input);
+
+                return new VectorValue(arrayPool: null, binaryEmbedding, binaryEmbedding);
+            }
+            case VectorEmbeddingType.Int8:
+            {
+                ReadOnlySpan<float> input;
+                if (array is float[] af)
+                    input = af;
+                else
+                    input = MemoryMarshal.Cast<T, float>(array);
+                
+                var int8Embedding = VectorQuantizer.ToInt8(input);
+                
+                return new VectorValue(arrayPool: null, (byte[])(object)int8Embedding, (byte[])(object)int8Embedding);
+            }
+            default:
+            {
+                if (typeof(T) == typeof(byte))
+                {
+                    return new VectorValue(arrayPool: null, (byte[])(object)array, (byte[])(object)array);
+                }
+                
+                var embeddings = (float[])(object)array;
+                var currentAllocator = Allocator ??= ArrayPool<byte>.Create();
+                int bytesRequires = embeddings.Length * sizeof(float); //todo store norm
+                byte[] buffer = currentAllocator.Rent(bytesRequires);
+                MemoryMarshal.Cast<float, byte>(embeddings).CopyTo(buffer);
+                return new VectorValue(currentAllocator, buffer, new Memory<byte>(buffer, 0, bytesRequires));
+            }
+        }
+    }
+    
+    private static VectorValue CreateSmartComponentsLocalEmbedding<TEmbedding>(in string text, in int dimensions)
```

Comment:
is this a correct name?

Reply:
No, changed into `CreateEmbeddingViaSmartComponentsLocalEmbedding`